### ISSUE TITLE
Web console terminal window too wide for mobile display

### ISSUE THREAD

**Issue by spadgett** [openshift/origin#6992]
Here is how it looks on an iPhone. Also it looks like the reconnect button is pushed onto the next line below the terminal and mostly offscreen.

![screenshot_2_3_16__8_29_am](https://cloud.githubusercontent.com/assets/1167259/12783574/58e5eae8-ca50-11e5-8485-a4d439c4a042.png)

/cc @jwforres @stefwalter

**Comment by jwforres** [comment]
the reconnect button being pushed off is a known bug https://bugzilla.redhat.com/show_bug.cgi?id=1276216

**Comment by jwforres** [comment]
Resizing the terminal window is a general problem, i think we were waiting on some API plumbing to be able to send SIGWINCH signal through to the container

**Comment by spadgett** [comment]
Can we size it initially so it's not larger than the device width?

**Comment by benjaminapetersen** [comment]
I think we can, but we would need to update the `kubernetes-container-terminal` to let us set attributes for `rows` and `cols`:

```javascript
// api supports sizing, at least initially.
// kubernetes-container-terminal/dist/container-terminal.js lines 90-95
var term = new Terminal({
    cols: 80,
    rows: 24,
    screenKeys: true
});
```
So if we update the directive `kubernetes-container-terminal rows="rowVal" cols="colVal"` we can at least set it via the controller.  We have a `BREAKPOINTS` constant in `app.js` now, could use that to set some standard `cols` vals based on the current screen width.

**Comment by spadgett** [comment]
Should we hide the terminal on mobile? I'm trying to imagine using it on an iPhone. The keyboard might cover most of the terminal.

**Comment by benjaminapetersen** [comment]
Yeah, it could be a bit silly to keep it if its not usable :smile:

Is the work to update the rows/cols valuable outside of mobile?

**Comment by jwforres** [comment]
We have a todo to make the terminal resize based on available space, but
the underlying api needs a fix to pass through the SIGWINCH signal,
otherwise even though visually the terminal would resize, it wouldn't
change the number of cols

**Comment by benjaminapetersen** [comment]
So we can't handle SIGWINCH (resize), but we should be able to handle opening to a size that matches the current viewport rather than a one-size-fits-all, right?

**Comment by jwforres** [comment]
Yeah we can specify an initial size.

**Comment by benjaminapetersen** [comment]
I'll look into doing something with the initial size.

**Comment by benjaminapetersen** [comment]
Have a PR open to [kubernetes-ui/container-terminal](https://github.com/kubernetes-ui/container-terminal/pull/3) to add attribs on the directive so we will be able to:
```html
<kubernetes-container-terminal
  pod="pod"
  container="container.name"
  prevent="!terminalTabWasSelected"
  rows="numRows"
  cols="numCols"
  screenKeys="true">
</kubernetes-container-terminal>
```
and then calc the rows/cols dynamically via breakpoints, etc.

@stefwalter ping! for review

**Comment by stefwalter** [comment]
Thanks @benjaminapetersen merged your pull request and released container-terminal 0.0.10

**Comment by benjaminapetersen** [comment]
Great!  

Only lingering issue is wether or not we should just hide it entirely on mobile, per @spadgett.  I'll experiment with setting it up to be small on mobile until we decide.

**Comment by spadgett** [comment]
The basic problem is fixed by #8149.

**Comment by jwforres** [comment]
We made the decision today to keep a minimum 80 cols x 124 rows in the terminal
